components/Seo.php

```diff
@@ -58,4 +36,152 @@ public function defineProperties()
             ]
         ];
     }
+
+    public function onRun()
+    {
+        $this->settings = Settings::instance();
+
+        if (!$this->page['viewBag']) {
+            $this->page['viewBag'] = new ViewBag();
+        }
+
+
+        if ($this->page->page->hasComponent('blogPost')) {
+            $post = $this->page['post'];
+            $properties = array_merge(
+                $this->page["viewBag"]->getProperties(),
+                $post->attributes,
+                $post->arcane_seo_options ?: []
+            );
+            $this->viewBagProperties = $properties;
+            $this->page['viewBag']->setProperties($properties);
+        } elseif (isset($this->page->apiBag['staticPage'])) {
+            $this->viewBagProperties = $this->page['viewBag'] = array_merge(
+                $this->page->controller->vars['page']->viewBag,
+                $this->page->attributes
+            );
+        } else {
+            $properties = array_merge(
+                $this->page['viewBag']->getProperties(), $this->page->settings
+            );
+
+            $this->viewBagProperties = $properties;
+            $this->page['viewBag']->setProperties($properties);
+        }
+        $this->disable_schema = $this->property('disable_schema');
+        $this->locale = App::getLocale();
```

Comment:
Get rid of the `locale` attribute. Instead, create a method that will retrieve the translated data. For example something like `getPropertyTranslated` which will internally use the locale. It will make the code a half shorter and easier to read.